src/components/button/button.ts

```diff
@@ -37,15 +55,15 @@ export class MdButton {
   /** Whether a mousedown has occurred on this element in the last 100ms. */
   isMouseDown: boolean = false;
 
-  setClassList() {return `md-${this.color}`;}
```

Comment:
I chatted with Misko and he agreed that this is an issue in Angular: https://github.com/angular/angular/issues/7289

In the meantime I would be okay with the renderer approach.